### ISSUE TITLE
Parser doesn't accept INSERT .. SELECT .. ON CONFLICT

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11427]
The following valid PostgreSQL query doesn't parse:

```sql
insert into "public"."t_author" ("id", "last_name") 
select ?, ? 
on conflict ("id") 
do update set "first_name" = ?
```

The error is:

```
Unsupported query type: [3:4] ..."."t_author" ("id", "last_name") 
select ?, ? 
on [*]conflict ("id") 
do update set "first_name" = ?
```

**Comment by lukaseder** [comment]
This is because `ON` is regarded as an alias in `SELECT`...

**Comment by lukaseder** [comment]
Yep, this parses just fine:

```sql
insert into "public"."t_author" ("id", "last_name") 
select ?, ? from (values (1)) t (v)
on conflict ("id") 
do update set "first_name" = ?
```

So, the issue is rather esoteric. Unlikely that someone would use the failing syntax rather than using `INSERT .. VALUES`.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.8 (#11428)
